chain/chain/src/store.rs

```diff
@@ -142,6 +142,20 @@ pub trait ChainStoreAccess {
     fn get_block_header(&mut self, h: &CryptoHash) -> Result<&BlockHeader, Error>;
     /// Returns hash of the block on the main chain for given height.
     fn get_block_hash_by_height(&mut self, height: BlockHeight) -> Result<CryptoHash, Error>;
+    /// Returns hash of the first available block after genesis.
+    fn get_earliest_block_hash(&mut self) -> Result<Option<CryptoHash>, Error> {
+        let head_header_height = self.head_header()?.height();
+        let tail = self.tail()?;
+        if let Ok(block_hash) = self.get_block_hash_by_height(tail) {
+            return Ok(Some(block_hash.clone()));
+        }
```

Comment:
This case is covered by the loop below. You can remove it safely.

Reply:
It does not cover the case when there is no block after the tail, though it is a rare corner case, maybe even impossible, though in the test suite we have a network with only the Genesis block and nothing else, so without this check it will return None 🤷‍♂️

What do you recommend?

Reply:
Good point. Could you please add a simple comment explaining this like "Case of having the only Block"?